### PR TITLE
Add vendorMarkup option and checkout price tooltip

### PR DESCRIPTION
**Pull request by danimoh**
Requires https://github.com/nimiq/vue-components/pull/81.

### CODE REVIEW

File: src/lib/CheckoutServerApi.ts [hunk 1]
```diff
@@ -141,6 +141,7 @@ export default class CheckoutServerApi {
                     currency: Currency.BTC,
                     type: PaymentType.DIRECT,
                     amount: '.0002e8',
+                    vendorMarkup: -.05,
```
Comment (sisou) [inline src/lib/CheckoutServerApi.ts:144]: Is this a different amount than in Demo.ts on purpose? And in the ETH option and the gasPrice?

File: src/lib/PublicRequestTypes.ts [hunk 2]
```diff
@@ -102,6 +102,10 @@ export interface PaymentOptions<C extends Currency, T extends PaymentType> {
      * i.e Luna for Currency.NIM and Satoshi for Currency.BTC
      */
     amount: string;
+    /**
+     * Crypto payment markup contained in `amount` that the vendor charges. Relative value, e.g. .01 for 1%.
+     */
+    vendorMarkup?: number;
```
Comment (sisou) [inline src/lib/PublicRequestTypes.ts:108]:
```suggestion
    /**
     * Crypto payment markup contained in `amount` that the vendor charges. Relative value, e.g. .01 for 1%.
     * Can be negative to describe a discount.
     */
    vendorMarkup?: number;
```

File: src/lib/CheckoutServerApi.ts (hunk 1 above)
Comment (danimoh) [reply src/lib/CheckoutServerApi.ts:144]: Yes, to test the update functionality.

**Comment by danimoh** [comment]
Will keep this as a draft until there's also a new Keyguard client with support for the vendor markup.

**Pull request merged**